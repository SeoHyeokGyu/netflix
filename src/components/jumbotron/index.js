import React from "react";

export default function Jumbotron(){
    return (
        <p>Jumba</p>
    );
}