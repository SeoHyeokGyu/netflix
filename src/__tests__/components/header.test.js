import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Header } from '../../components';

describe('<Header />', () => {
  it('renders the basic <Header /> with background', () => {
    const { container, getByText, getByTestId } = render(
      <Router>
        <Header>
          <Header.Frame>
            <Header.Logo src="/logo.svg" alt="Netflix" />
            <Header.TextLink active="true">Hello I am a Link!</Header.TextLink>
          </Header.Frame>
        </Header>
      </Router>
    );

    expect(getByText('Hello I am a Link!')).toBeTruthy();
    expect(getByTestId('header-bg')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the basic <Header /> without background', () => {
    const { container, getByText, queryByTestId } = render(
      <Router>
        <Header bg="false">
          <Header.Frame>
            <Header.Logo src="/logo.svg" alt="Netflix" />
            <Header.ButtonLink>Sign In</Header.ButtonLink>
            <Header.TextLink active="flase">Hello I am a Link!</Header.TextLink>
          </Header.Frame>
        </Header>
      </Router>
    );

    expect(getByText('Hello I am a Link!')).toBeTruthy();
    expect(queryByTestId('header-bg')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the full <Header> with a background', () => {
    const { container, getByText, getByTestId } = render(
      <Router>
        <Header src="joker1" dontShowOnSmallViewPort>
          <Header.Frame>
            <Header.Group>
              <Header.Logo src="/images/logo.svg" alt="Netflix" />
              <Header.TextLink active={false} onClick={() => {}}>
                Series
              </Header.TextLink>
              <Header.TextLink active onClick={() => {}}>
                Films
              </Header.TextLink>
            </Header.Group>
            <Header.Group>
              <Header.Search searchTerm="Joker" setSearchTerm={() => {}} />
              <Header.Profile>
                <Header.Picture src="/images/karl.png" />
                <Header.Dropdown>
                  <Header.Group>
                    <Header.Picture src="/images/karl.png" />
                    <Header.TextLink>Karl Hadwen</Header.TextLink>
                  </Header.Group>
                  <Header.Group>
                    <Header.TextLink onClick={() => {}}>Sign out</Header.TextLink>
                  </Header.Group>
                </Header.Dropdown>
              </Header.Profile>
            </Header.Group>
          </Header.Frame>

          <Header.Feature>
            <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
            <Header.Text>Forever alone in a crowd...</Header.Text>
            <Header.PlayButton>Play</Header.PlayButton>
          </Header.Feature>
        </Header>
      </Router>
    );

    expect(getByTestId('search-input')).toBeTruthy();
    expect(getByTestId('search-input').value).toBe('Joker');
    fireEvent.change(getByTestId('search-input'), { target: { value: 'Simpsons' } });
    fireEvent.click(getByTestId('search-click'));

    expect(getByText('Series')).toBeTruthy();
    expect(getByText('Films')).toBeTruthy();
    expect(getByText('Karl Hadwen')).toBeTruthy();
    expect(getByText('Watch Joker Now')).toBeTruthy();
    expect(getByText('Sign out')).toBeTruthy();
    expect(getByText('Forever alone in a crowd...')).toBeTruthy();
    expect(getByText('Play')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
