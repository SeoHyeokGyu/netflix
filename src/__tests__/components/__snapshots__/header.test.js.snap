// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic <Header /> with background 1`] = `
<div
  class="header__Background-djltcn-0 jsAUyh"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-11 jmlSof"
  >
    <a
      href=""
    >
      <img
        alt="Netflix"
        class="header__Logo-djltcn-12 kZoHXb"
        src="/logo.svg"
      />
    </a>
    <p
      class="header__Link-djltcn-7 gMXjBj"
    >
      Hello I am a Link!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the basic <Header /> without background 1`] = `
<div
  class="header__Background-djltcn-0 jsAUyh"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-11 jmlSof"
  >
    <a
      href=""
    >
      <img
        alt="Netflix"
        class="header__Logo-djltcn-12 kZoHXb"
        src="/logo.svg"
      />
    </a>
    <a
      class="header__ButtonLink-djltcn-13 ciYHjp"
      href=""
    >
      Sign In
    </a>
    <p
      class="header__Link-djltcn-7 kcIbdf"
    >
      Hello I am a Link!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the full <Header> with a background 1`] = `
<div
  class="header__Background-djltcn-0 kEkyAf"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-djltcn-11 jmlSof"
  >
    <div
      class="header__Group-djltcn-8 jFzvTm"
    >
      <a
        href=""
      >
        <img
          alt="Netflix"
          class="header__Logo-djltcn-12 kZoHXb"
          src="/images/logo.svg"
        />
      </a>
      <p
        class="header__Link-djltcn-7 kcIbdf"
      >
        Series
      </p>
      <p
        class="header__Link-djltcn-7 kcIbdf"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-djltcn-8 jFzvTm"
    >
      <div
        class="header__Search-djltcn-3 eFrNaT"
      >
        <button
          class="header__SearchIcon-djltcn-4 hnCjKu"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-djltcn-5 bGTGcG"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-djltcn-10 hmGsyV"
      >
        <button
          class="header__Picture-djltcn-6 bUcDpL"
          src="/images/users//images/karl.png.png"
        />
        <div
          class="header__Dropdown-djltcn-9 btDXDW"
        >
          <div
            class="header__Group-djltcn-8 jFzvTm"
          >
            <button
              class="header__Picture-djltcn-6 bUcDpL"
              src="/images/users//images/karl.png.png"
            />
            <p
              class="header__Link-djltcn-7 kcIbdf"
            >
              Karl Hadwen
            </p>
          </div>
          <div
            class="header__Group-djltcn-8 jFzvTm"
          >
            <p
              class="header__Link-djltcn-7 kcIbdf"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-djltcn-11 header__Feature-djltcn-14 jmlSof izBabp"
  >
    <h2
      class="header__FeatureCallOut-djltcn-16 YlHHm"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-djltcn-15 hRFeXs"
    >
      Forever alone in a crowd...
    </p>
    <button
      class="header__PlayButton-djltcn-2 bvcChh"
    >
      Play
    </button>
  </div>
</div>
`;
